src/shell/Shell.cpp

```diff
@@ -328,17 +227,13 @@ static Trap::TrapResult executeWASM(Store* store, const std::string& filename, c
                     nullptr));
             }
         } else if (import->moduleName() == "wasi_snapshot_preview1") {
-            // TODO wasi
-            if (import->fieldName() == "proc_exit") {
-                auto ft = functionTypes[SpecTestFunctionTypes::I32];
+            Walrus::WASI::WasiFunc* wasiImportFunc = wasi->find(import->fieldName());
+            FunctionType* fn = functionTypes[wasiImportFunc->functionType];
```

Comment:
When `find` method of WASI cannot get a valid wasi function, `wasiImportFunc` has a null pointer which leads to invalid memory access (null pointer dereference) in `wasiImportFunc->functionType`.

Reply:
Should be fixed now to avoid null ptr dereference.